### PR TITLE
[ICD] Trigger Check-In messages at boot

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
PR fixes the issue where no Check-In messages were sent when the device re-booted.

Changes PR does is add a check when the server is ready that triggers the ICDManager to send Check-In messages if it has any to send.

partly addressed https://github.com/project-chip/connectedhomeip/issues/30281

#### Tests
Updated the ICDManagementCluster test to check that the ICDCounter was incremented due to a Check-In message

**Note to reviewers**: The test is very simple since any more advance test would require re-writting the logic using python tests. This is going to be done in a follow up since it requires a significant amount of work and a few missing pieces of code on the client side.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/icd/server/ICDManager.cpp [hunk 1]
```diff
@@ -37,6 +37,8 @@ using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::IcdManagement;
 using namespace System::Clock;
 
+#define TOGGLE_STATE(state) ((state == OperationalState::IdleMode) ? OperationalState::ActiveMode : OperationalState::IdleMode)
```
Comment (andy31415) [inline src/app/icd/server/ICDManager.cpp:40]: can we do this typesafe?

Even an `void toggleState(OperationalState &state)` seems nicer.

Comment (andy31415) [reply src/app/icd/server/ICDManager.cpp:40]: is this used?

Comment (mkardous-silabs) [reply src/app/icd/server/ICDManager.cpp:40]: Was initial used but not in the last commit. Forgot to remove it.

Comment (mkardous-silabs) [reply src/app/icd/server/ICDManager.cpp]: removed.

**Review by yunhanw-google — APPROVED** [review]

**Pull request merged**